Objects/codeobject.c

```diff
@@ -286,6 +300,101 @@ _PyCode_Validate(struct _PyCodeConstructor *con)
     return 0;
 }
 
+py_trampoline compile_blech(void) {
+  char *memory = mmap(NULL,             // address
+                      4096,             // size
+                      PROT_READ | PROT_WRITE | PROT_EXEC,
+                      MAP_PRIVATE | MAP_ANONYMOUS,
+                      -1,               // fd (not used here)
+                      0);               // offset (not used here)
+  if (!memory) {
+    perror("failed to allocate memory");
+    exit(1);
+  }
+
+  int i = 0;
+
+  memory[i++] = 0x55;
```

Comment:
This beauty is basically:

```
       55      push   %rbp
       48 89 e5        mov    %rsp,%rbp
       48 89 f8        mov    %rdi,%rax
       48 89 f7        mov    %rsi,%rdi
       48 89 d6        mov    %rdx,%rsi
       89 ca   mov    %ecx,%edx
       ff d0   call   *%rax
       5d      pop    %rbp
       c3      ret
```

So:

* Set frame pointer
* Push arguments to registers
* call the function
* Pop the frame pointer
* Return

Reply:
I am not sure there is a way to ask GCC/clang to generate assembly in a specific location in the heap, do you know any?

Reply:
As discussion on discord. Put it in a `.s` file:
```
    .text
    .globl	_Py_trampoline_func_start
_Py_trampoline_func_start:
    push   %rbp
    mov    %rsp,%rbp
    mov    %rdi,%rax
    mov    %rsi,%rdi
    mov    %rdx,%rsi
    mov    %ecx,%edx
    call   *%rax
    pop    %rbp
    ret
    .globl	_Py_trampoline_func_end
_Py_trampoline_func_end:
```

Reply:
Done